xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/PVR.h

```diff
@@ -0,0 +1,7939 @@
+/// @ref cpp_kodi_addon_pvr_Timers "Timers methods in PVR instance class".
+///
+//------------------------------------------------------------------------------
+
+//##############################################################################
+/// @defgroup cpp_kodi_addon_pvr_Defs_Menuhook 7. Menuhook
+/// @brief **PVR Context menu data**\n
+/// Define data for the context menus available to the user
+///
+//------------------------------------------------------------------------------
+
+//##############################################################################
+/// @defgroup cpp_kodi_addon_pvr_Defs_EDLEntry 8. Edit definition list (EDL)
+/// @brief **An edit decision list or EDL is used in the post-production process
+/// of film editing and video editing**\n
+/// Used on @ref kodi::addon::CInstancePVRClient::GetEPGTagEdl and
+/// @ref kodi::addon::CInstancePVRClient::GetRecordingEdl
+///
+//------------------------------------------------------------------------------
+
+//##############################################################################
+/// @defgroup cpp_kodi_addon_pvr_Defs_Stream 9. Inputstream
+/// @brief **Inputstream**\n
+/// This includes classes and values that are used in the PVR inputstream.
+///
+/// Used on @ref cpp_kodi_addon_pvr_Streams "Inputstream methods in PVR instance class".
+///
+/// @note The parts here will be removed in the future and replaced by the
+/// separate @ref cpp_kodi_addon_inputstream "inputstream addon instance".
+/// If there is already a possibility, new addons should do it via the
+/// inputstream instance.
+///
+//------------------------------------------------------------------------------
+
+//##############################################################################
+/// @defgroup cpp_kodi_addon_pvr_Defs_Various 10. Various
+/// @brief **Various definitions, structures and enumerators**\n
+/// Here are values that cannot be assigned directly to a group.
+///
+//------------------------------------------------------------------------------
+
+//@}
+
+//}}}
+//______________________________________________________________________________
+//--==----==----==----==----==----==----==----==----==----==----==----==----==--
+//¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+// "C++" Definitions group 1 - General PVR
+//{{{
+
+//==============================================================================
+/// @defgroup cpp_kodi_addon_pvr_Defs_PVRCapabilities class PVRCapabilities
+/// @ingroup cpp_kodi_addon_pvr_Defs_General
+/// @brief **PVR add-on capabilities**\n
+/// This class is needed to tell Kodi which options are supported on the addon.
+///
+/// If a capability is set to **true**, then the corresponding methods from
+/// @ref cpp_kodi_addon_pvr "kodi::addon::CInstancePVRClient" need to be
+/// implemented.
+///
+/// Used on @ref kodi::addon::CInstancePVRClient::GetCapabilities().
+///
+//@{
+class PVRCapabilities
```

Comment:
Wouldn't it make sense to have a separate header file for each of the c++ classes instead of making one big fat PVR.h? If I only need a channel, why should I include everything? Slows down compilation time for no good.

Reply:
Thought also about, but not so easy. Everything goes to `kodi::addon::CAddonInstancePVR` and some parts takes structures/values from another part, where then a hen/egg problem on headers :smirk:.

Hope this `///{{{` code folding works on your IDE, where really, really helpfull about.

Reply:
I hate code folding. I also want to see everything that is actually in the file. Please, do not publish those monster header!!! Make it nice and clean - small files with one responsibility it is! Or do you want to create the next generation `windows.h`?